contrib/create-waku/src/cli.ts

```diff
@@ -0,0 +1,191 @@
+        },
+        {
+          name: "overwriteChecker",
+          type: (values: any) => {
+            if (values === false) {
+              throw new Error(red("✖") + " Operation cancelled");
+            }
+            return null;
+          },
+        },
+        {
+          name: "packageName",
+          type: () => (isValidPackageName(targetDir) ? null : "text"),
+          message: "Package name",
+          initial: () => toValidPackageName(targetDir),
+          validate: (dir: string) =>
+            isValidPackageName(dir) || "Invalid package.json name",
+        },
+        {
+          name: "chooseProject",
+          type: "select",
+          message: "Choose a starter template",
+          choices: [
+            { title: "basic-template", value: CHOICES[0] },
+            { title: "async-template", value: CHOICES[1] },
+            { title: "promise-template", value: CHOICES[2] },
+          ],
+        },
+      ],
+      {
+        onCancel: () => {
+          throw new Error(red("✖") + " Operation cancelled");
+        },
+      }
+    );
+  } catch (cancelled) {
+    if (cancelled instanceof Error) {
+      console.log(cancelled.message);
+    }
+    process.exit(1);
+  }
+
+  const { packageName, shouldOverwrite, chooseProject } = result;
+
+  const root = path.resolve(targetDir);
+
+  if (shouldOverwrite) {
+    fse.emptyDirSync(root);
+  } else if (!fs.existsSync(root)) {
+    fs.mkdirSync(root);
+  }
+
+  const pkg = {
+    name: packageName ?? toValidPackageName(targetDir),
+    version: "0.0.0",
+  };
+
+  fs.writeFileSync(
+    path.join(root, "package.json"),
+    JSON.stringify(pkg, null, 2)
+  );
+
+  console.log("Setting up project...");
+  const templateRoot = path.join(__dirname, "template");
+
+  const render = function render(templateName: string) {
+    const templateDir = path.resolve(templateRoot, templateName);
+
+    // Read existing package.json from the root directory
+    const existingPackageJsonPath = path.join(root, "package.json");
+    const existingPackageJson = JSON.parse(
+      fs.readFileSync(existingPackageJsonPath, "utf-8")
+    );
+
+    // Read new package.json from the template directory
+    const newPackageJsonPath = path.join(templateDir, "package.json");
+    const newPackageJson = JSON.parse(
+      fs.readFileSync(newPackageJsonPath, "utf-8")
+    );
+
+    const mergedPackageJson = {
+      ...newPackageJson,
+      ...existingPackageJson,
+    };
+
+    fse.copySync(templateDir, root);
+
+    // Write the merged package.json back to the target directory
+    const existingPackageJson = JSON.parse(
+      fs.readFileSync(existingPackageJsonPath, "utf-8")
+    );
+    fs.writeFileSync(
+      existingPackageJsonPath,
+      JSON.stringify({
+        ...existingPackageJson,
+        name: packageName ?? toValidPackageName(targetDir),
+        version: "0.0.0",
+      }, null, 2)
+    );
+  };
+
+  render(chooseProject as string);
```

Comment:
It's probably inferred correctly.
```suggestion
  render(chooseProject);
```